app/models/student_application.rb

```diff
@@ -11,8 +11,6 @@
 #
 
 class StudentApplication < ActiveRecord::Base
-  include AssignToCurrentSemester
```

Comment:
Got it. I totally agree with this change because of the issues you mentioned!

Although it was a sexy solution haha